### PR TITLE
[feature] Add travis check to make sure we're properly managing go mods

### PR DESCRIPTION
**Pull request by edulop91**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,15 +17,28 @@ install:
...
+        - STEP_SPAN_ID=$(echo script | sum | cut -f 1 -d \ )
+        - buildevents cmd $TRAVIS_BUILD_ID $STEP_SPAN_ID codecov-validate -- curl --data-binary @codecov.yml https://codecov.io/validate
+        - buildevents cmd $TRAVIS_BUILD_ID $STEP_SPAN_ID make-test-coverage -- make test-coverage
+        # Disable snyk until https://github.com/snyk/snyk/issues/354 is resolved
+        # - if [[ ! -z "${SNYK_TOKEN}" ]]; then snyk monitor --org=czi; snyk test; fi
+        - buildevents step $TRAVIS_BUILD_ID $STEP_SPAN_ID $STEP_START script
+    - stage: check
+      script: |
+        set -e
+        export GO111MODULE=on
```
Comment (ryanking) [inline .travis.yml]: shouldn't we set this globally?

Comment (edulop91) [reply .travis.yml]: Hmm yeah, I tried that and something else broke. Let me try again.

Comment (ryanking) [reply .travis.yml]: Not a big deal, don't spend too much time on it

**Comment by edulop91** [comment]
```
--- FAIL: TestCompareLocalAndGlobal (0.21s)
    require.go:794: 
        	Error Trace:	latest_test.go:22
        	Error:      	Received unexpected error:
        	            	Could not generate url for github.com/chanzuckerberg/cztack//aws-params-reader-policy?ref=v0.15.1 latest module: GET https://api.github.com/repos/chanzuckerberg/cztack/releases/latest: 403 API rate limit exceeded for 34.66.178.120. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.) [rate reset in 27m06s]
        	Test:       	TestCompareLocalAndGlobal
```
